### PR TITLE
Dlq smaple

### PR DESCRIPTION
**Pull request by JDragovichAlertLogic**
### Problem Description
This PR samples teh first 5 messages from a collectors dlq and stringifies them to send with teh checkin body.

### CODE REVIEW

File: dlblob.js [hunk 1]
```diff
@@ -78,13 +79,20 @@ class AlAzureDlBlob {
...
                 } else {
-                    const dlstats =  {
-                        dl_stats: {
-                            dl_count: dlblobList.entries.length,
-                            max_dl_size: dlblob._findMaxDlBlobSize(dlblobList.entries)
-                        }
-                    };
-                    return callback(null, dlstats);
+                    const sample = dlblobList.entries.slice(0,DLQ_SAMPLE_SIZE);
+                    async.mapLimit(sample, CONCURRENT_BLOB_PROCESS_NUM, async.reflect(function(blob, asyncCallback) {
```
Comment (kkuzmin) [inline dlblob.js:83]: please add a test when `sample` is an empty list

File: dlblob.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ const async = require('async');
 const azureStorage = require('azure-storage');
 
 const CONCURRENT_BLOB_PROCESS_NUM = 20;
+const DLQ_SAMPLE_SIZE = 5;
```
Comment (kkuzmin) [inline dlblob.js]: what is an average size of one DL blob? Five can be too mush for checkin body. Maybe start with 1 or 2

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: test/dlblob_test.js [hunk 3]
```diff
@@ -97,6 +98,143 @@ describe('Dead letter blob processing unit tests.', function() {
...
+        .get('/alertlogic-dl')
+        .query({'restype':'container','comp':'list','prefix':process.env.WEBSITE_SITE_NAME})
+        .times(5)
+        .reply(200, (uri) => {
+            return mock.LIST_CONTAINER_BLOBS_EMPTY()
+        });
+        
+        // Get blob content
+        var getBlobTextStub = sinon.fake();
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
```
Comment (kkuzmin) [inline test/dlblob_test.js]: we can remove this nock because it should not be used

File: test/dlblob_test.js [hunk 4]
```diff
@@ -97,6 +98,143 @@ describe('Dead letter blob processing unit tests.', function() {
...
+        .get(/alertlogic-dl.*/)
+        .times(6)
+        .reply(200, () => {
+            getBlobTextStub();
+            return JSON.stringify(mock.GET_BLOB_CONTENT_TEXT)
+        });
+        
+        // Delete blob
+        var deleteBlobStub = sinon.fake();
```
Comment (kkuzmin) [inline test/dlblob_test.js]: we can remove this nock because it should not be used

**Review by kkuzmin — CHANGES_REQUESTED** [review]

**Review by kkuzmin — APPROVED** [review]
